modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/tasks/WatchTask.java

```diff
@@ -180,63 +194,50 @@ public void watch(IncrementalTaskInputs incrementalTaskInputs)
 
 		Logger logger = getLogger();
 
-		long bundleId = -1;
+		long installedBundleId = -1;
 
-		File outputFile = getOutputFile();
+		try (GogoShellClient gogoShellClient = new GogoShellClient()) {
+			installedBundleId = _getInstalledBundleId(gogoShellClient);
 
-		if (outputFile.exists()) {
-			try (GogoShellClient gogoShellClient = new GogoShellClient()) {
-				File manifestFile = new File(
-					getBundleDir(), "META-INF/MANIFEST.MF");
+			if ((installedBundleId < 1) ||
+				!incrementalTaskInputs.isIncremental()) {
 
-				String bundleSymbolicName = FileUtil.readManifestAttribute(
-					manifestFile, Constants.BUNDLE_SYMBOLICNAME);
+				_installOrUpdateBundle(installedBundleId, gogoShellClient);
 
-				long installedBundleId = _getBundleId(
-					bundleSymbolicName, gogoShellClient);
+				return;
+			}
 
-				String outputFileBundleIdString = new String(
-					Files.readAllBytes(outputFile.toPath()),
-					StandardCharsets.UTF_8);
+			List<File> modifiedFiles = _getModifiedFiles(incrementalTaskInputs);
 
-				long outputFileBundleId = Long.parseLong(
-					outputFileBundleIdString);
+			if (_isClassLoaderFileChanged(modifiedFiles) ||
+				_isManifestChanged(modifiedFiles)) {
 
-				if (installedBundleId == outputFileBundleId) {
-					bundleId = outputFileBundleId;
-				}
+				_refreshBundle(installedBundleId, gogoShellClient);
+
+				return;
 			}
-			catch (Exception e) {
-				if (logger.isDebugEnabled()) {
-					logger.debug("Unable to get bundle ID", e);
+
+			if (logger.isQuietEnabled()) {
+				if (modifiedFiles.isEmpty()) {
+					logger.quiet("No files changed. Skipping bundle refresh.");
+				}
+				else {
+					logger.quiet(
+						"Only resources changed. Skipping bundle refresh.");
 				}
 			}
 		}
+	}
 
-		if ((bundleId < 1) || !incrementalTaskInputs.isIncremental()) {
-			_installBundleFully();
-
-			return;
-		}
-
-		List<File> modifiedFiles = _getModifiedFiles(incrementalTaskInputs);
-
-		if (_isClassLoaderFileChanged(modifiedFiles) ||
-			_isManifestChanged(modifiedFiles)) {
+	private static Attributes _getAttributes(File bundleDir)
```

Comment:
This logic is only used once.  Adding it directly to `L504` won't hurt readability and it removes a method.